docs/installation/docker-compose.md

```diff
@@ -0,0 +1,299 @@
+
+If you are stuck or find issues with the installation process you can either file an issue on our Github repository, or join our Slack [here](https://autolab-slack.herokuapp.com/) and let us know and we will try our best to help. Also see the [debugging](#debugging-your-deployment) section for tips on how to diagnose problems.
+
+## Installation
+First ensure that you have Docker and Docker Compose installed on your machine. See the official [Docker docs](See https://docs.docker.com/install/) for the installation steps.
+
+1. Clone this repository and its Autolab and Tango submodules: 
+
+        :::bash
+        git clone --recurse-submodules -j8 git://github.com/autolab/docker.git autolab-docker
+
+
+2. Enter the project directory:
+
+        :::bash
+        cd autolab-docker
+
+3. Update our Autolab and Tango submodules to ensure that you are getting the latest versions:
+
+        :::bash
+        make update
+
+4. Create initial default configs:
+
+        :::bash
+        make
+
+5. Build the Dockerfiles for both Autolab and Tango:
+
+        :::bash
+        docker-compose build
+
+6. Run the Docker containers:
+
+        :::bash
+        docker-compose up -d
+
+    Note at this point Nginx will still be crash-looping in the Autolab container because SSL has not been configuired/disabled yet.
+
+7. Ensure that the newly created config files have the right permissions, as it may have been modified during the building process:
+
+        :::bash
+        make set-perms
+
+8. Perform database migrations for Autolab, which will initialize your database schema:
+
+        :::bash
+        make db-migrate
+
+9. Create administrative user for Autolab:
+
+        :::bash
+        make create-user
+
+    This user has full permissions on Autolab and will be able to create other users and designate other admins.
+
+10. Change `DOCKER_TANGO_HOST_VOLUME_PATH` in `docker-compose.yml` to be the absolute path to the Tango `volumes` directory, i.e `/<path-to-docker-compose-installation>/Tango/volumes`. This is so that Tango knows where to put the output files of its autograded jobs.
+
+        :::bash 
+        # in docker-compose.yml
+
+        # Modify the below to be the path to volumes on your host machine
+        - DOCKER_TANGO_HOST_VOLUME_PATH=/home/your-user/autolab-docker/Tango/volumes # example path
+
+11. Stop all containers, as we are going to setup/disable TLS:
+
+        :::bash 
+        docker-compose stop
+
+12. If you intend to use TLS later, in `nginx/app.conf`, change instances of `<REPLACE_WITH_YOUR_DOMAIN>` to your real domain name. Otherwise, if you are not using TLS, in `nginx/no-ssl-app.conf`, change `server_name` to your real domain name.
+
+13. Continue with TLS setup as outlined in the next section
```

Comment:
Add a hyper link to the next section

Reply:
recommend to call it SSL/TLS

Reply:
I will just mention SSL/TLS in the first instance so that people will be aware that they are synonymous, and afterwards use TLS to aid the efforts of the cryptographic community to use the right name as SSL is technically already deprecated